src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -44,22 +44,27 @@ def __init__(self, pdict, verbose=None, logger=None, validRSEs=None):
             validRSEs = []
         self.validRSEs = validRSEs
 
+        expectedRSEs = pdict['expectedRSEs']
+        if len(expectedRSEs) == 0:
+            msg = 'MSPileup document require non-empty list of expectedRSEs'
+            raise Exception(msg)
+
         self.data = {
-            'pileupName': pdict.get('pileupName', ''),
-            'pileupType': pdict.get('pileupType', ''),
+            'pileupName': pdict['pileupName'],
+            'pileupType': pdict['pileupType'],
             'insertTime': pdict.get('insertTime', gmtimeSeconds()),
             'lastUpdateTime': pdict.get('lastUpdateTime', gmtimeSeconds()),
-            'expectedRSEs': pdict.get('expectedRSEs', []),
+            'expectedRSEs': expectedRSEs,
             'currentRSEs': pdict.get('currentRSEs', []),
-            'fullReplicas': pdict.get('fullReplicas', 0),
+            'fullReplicas': pdict.get('fullReplicas', 1),
             'campaigns': pdict.get('campaigns', []),
             'containerFraction': pdict.get('containerFraction', 1.0),
-            'replicationGrouping': pdict.get('replicationGrouping', ""),
+            'replicationGrouping': pdict.get('replicationGrouping', 'ALL'),
             'activatedOn': pdict.get('activatedOn', gmtimeSeconds()),
             'deactivatedOn': pdict.get('deactivatedOn', gmtimeSeconds()),
             'active': pdict.get('active', False),
```

Comment:
given that it's mandatory, you might want to keep `active` in sync with the other keys, hence, direct access to that key (pdict['active'])

Reply:
done